R/retrieveData-KNMI-OpenAPI.R

```diff
@@ -0,0 +1,218 @@
+                               toupper(), 
+                             "_____L3__", 
+                             stringr::str_split_1(start_date_seq[.x], pattern = "-")[1], 
+                             stringr::str_split_1(start_date_seq[.x], pattern = "-")[2],
+                             stringr::str_split_1(start_date_seq[.x], pattern = "-")[3],
+                             knmi_var_lookup %>% 
+                               dplyr::filter(var_name == variable) %>% 
+                               dplyr::pull("filename_timestamp"), "_",
+                             stringr::str_split_1(end_date_seq[.x], pattern = "-")[1], 
+                             stringr::str_split_1(end_date_seq[.x], pattern = "-")[2],
+                             stringr::str_split_1(end_date_seq[.x], pattern = "-")[3],
+                             knmi_var_lookup %>% 
+                               dplyr::filter(var_name == variable) %>% 
+                               dplyr::pull("filename_timestamp"), "_",
+                             
+                             if(lubridate::as_date(start_date_seq[.x]) < lubridate::as_date("2018-01-01")) {
+                               
+                               knmi_var_lookup %>% 
+                                 dplyr::filter(var_name == variable) %>% 
+                                 dplyr::pull("file_ending_before2018")
+                               
+                             } else {
+                               
+                               "0006.nc"
+                               
+                             }
+    )
+  },
+  .progress = TRUE) 
+  
+  file_names <- do.call(rbind, file_name)
+  
+  
+  # GET request KNMI Open API
+  purrr::map(.x = file_names,
+             .f = ~{
+               
+               knmi_api <- httr::GET(url = paste0("https://api.dataplatform.knmi.nl/open-data/v1/datasets/",
+                                                  knmi_var_lookup %>%
+                                                    dplyr::filter(var_name == variable) %>%
+                                                    dplyr::pull("collection"),
+                                                  "/versions/", version, "/files/",
+                                                  .x, "/url"),
+                                     httr::add_headers(Authorization = knmi_open_key)) 
+               
+               # print warning if GET request failes
+               if(!rlang::is_empty(jsonlite::fromJSON(txt = rawToChar(x = knmi_api$content))$message)) {
+                 message(paste0("API failed to retrieve data for the file name: ", .x))
+                 }
+               
+               # dowload netCDF file per day
+               download.file(url = jsonlite::fromJSON(rawToChar(knmi_api$content))$temporaryDownloadUrl,
+                             destfile = here::here("data", "netCDF files", paste0(knmi_var_lookup %>% 
```

Comment:
This hard-coded destination does not (necessarily) match with process-KNMI-netCDF-files.R script. To match this, we should add arguments to this function.

Reply:
Alternatively, as I referred to in a comment on the process script, we could also circumvent this by forcing the function to download the files to a temporary folder, and then at the end of the process function, this temporary folder is removed/cleaned. What do you think about this?

Only if we expect that people want to have a closer look at the individual netCDF files, there is no need to retain them after they are processed, I think.

Reply:
Unless they would want to change the bounding box slightly, they then would need to download all the files again, if they are removed. 
I guess, it is not super unlikely that you would maybe want to adapt the bounding box, that was the reason I did not include removing the files yet.